frontend/webadmin/modules/gwt-common/src/main/resources/org/ovirt/engine/core/Common.gwt.xml

```diff
@@ -311,6 +311,7 @@
         <include name="common/businessentities/storage/LunStatus.java" />
         <include name="common/businessentities/storage/PropagateErrors.java" />
         <include name="common/businessentities/storage/RepoImage.java" />
+        <include name="common/businessentities/storage/ScratchDiskInfo.java" />
```

Comment:
Ah I see, when I wrote that comment I thought (not sure why) that `ScratchDiskInfo` is created by StartVmBackup and only used by the vds command.. so why not doing it that way?  CreateScratchDisk and RemoveScratchDisk don't need the path, only StartVmBackup needs to set it for VmBackupConfigVDSCommandBase, right?
AFAIR, prepare-image is a synchronous operation and if so we don't need two phases in StartVmBackup but rather to call prepare image and start the backup right after, without the need to store the prepared-paths in the database (using the parameters) - does it make sense?

Reply:
`ScratchDiskInfo` is first created by `CreateScratchDisksCommand` (during `CREATING_SCRATCH_DISKS` phase).
The path is initialized by `StartVmBackupCommand::prepareScratchDisks()` (during the `PREPARING_SCRATCH_DISK` phase that immediately follows the above `CREATING_SCRATCH_DISKS` phase).
Do you suggest to unite those phases?

Reply:
Yes but not these two because 'prepare' should be executed on the created disks and create-disk is an asynchronous operation, so to unite `PREPARING_SCRATCH_DISK` and `STARTING`

Reply:
So let me summarize: you want me eliminate `STARTING` backup phase as totally redundant.
That means `CREATING_SCRATCH_DISKS` --> `PREPARING_SCRATCH_DISK` --> `READY`?
Do I understand you correctly?
If so I guess it's better be done in a separate PR. Because there are few more files including `api.model`.
Anyway I had some more thoughts about some code refactoring there in the state machine.

Reply:
@barpavel please do not make any change to backup phases. We want to add scratch disk
uuids in backup flow - this change must not change backup flow. We can improve 
backup flows later if needed.

Reply:
@nirs After talking to Arik, AFAIU him, he suggested to unite 2 phases: `PREPARING_SCRATCH_DISK` & `STARTING` into `STARTING`.

One of the reasons of Arik's suggestion was to try not to expose the new `ScratchDiskInfo` into the `Common.gwt.xml` file by putting the new class (or it's content) inside the `VmBackupConfigVDSCommandBase` class (that is a base class of `StartVmBackupVDSCommand` invoked by `StartVmBackup` command).

Putting aside the suggestion to unite the 2 phases, @ahadas I don't see how it can be achieved.
The new `ScratchDiskInfo` class is used in `VmBackupParameters` to pass configuration between `CREATING_SCRATCH_DISKS` and `PREPARING_SCRATCH_DISK` phases (the phases that **can not be united**), so moving the `ScratchDiskInfo` class into `VmBackupConfigVDSCommandBase` besides being a little bit awkward (started the code refactoring and saw that it's not that elegant to say the least) will not help here. In this case either `VmBackupConfigVDSCommandBase` class and/or its inner `static ScratchDiskInfo` class will still need to be part of the `org.ovirt.engine.core.common.action.VmBackupParameters` class, thus we'll be back to the original problem that was caused by being all the classes under `action`  (as `VmBackupParameters` above) being compiled.
See `Common.gwt.xml` class `<include name="common/action/*.java" />` rule.

Sorry for not that readable answer, @ahadas If I misunderstood you please tell me or let's have an offline chat about that :)

Reply:
@barpavel I wrote [a sketch of what I had in mind](https://github.com/ahadas/ovirt-engine/commit/8bbeeab8c66cb710a96e734d9558c306df4d3574) - didn't even compile it though, can you please take a look and see if something like that can work?

Reply:
@ahadas As you already noticed I fixed according to your suggestions :)

Reply:
Also I believe it will be easier to review it commit after commit, not all files together :)